src/OrleansRuntime/Placement/PlacementDirectorsManager.cs

```diff
@@ -1,46 +1,43 @@
 using System;
-using System.Collections.Generic;
 using System.Threading.Tasks;
-using Orleans.Runtime.Configuration;
 
 namespace Orleans.Runtime.Placement
 {
+    using System.Collections.Concurrent;
+
+    using Microsoft.Extensions.DependencyInjection;
+
     internal class PlacementDirectorsManager
     {
-        private readonly Dictionary<Type, PlacementDirector> directors = new Dictionary<Type, PlacementDirector>();
-        private PlacementStrategy defaultPlacementStrategy;
-        private ClientObserversPlacementDirector clientObserversPlacementDirector;
+        private readonly ConcurrentDictionary<Type, IPlacementDirector> directors = new ConcurrentDictionary<Type, IPlacementDirector>();
```

Comment:
I tend to favor 'obviously not deficient' over 'not obviously deficient', so ConcurrentDictionary whenever I'm not completely certain that something will only ever be accessed in a single-threaded fashion. If you prefer, I can put a lock on it and use regular Dictionary